sw/device/tests/rv_core_ibex_rnd_test.c

```diff
@@ -0,0 +1,99 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_edn.h"
+#include "sw/device/lib/dif/dif_rv_core_ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/entropy_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "rv_core_ibex_regs.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+dif_result_t rv_core_ibex_read_rnd_data_followed_by_status(
+    const dif_rv_core_ibex_t *rv_core_ibex, uint32_t *data,
```

Comment:
I would make this funcion return void and use the `CHECK` macro inside to verify whether the test has failed. It is easier to debug as the `CHECK` indicates the line that failed.